src/state/opamEnv.ml

```diff
@@ -267,11 +328,11 @@ let apply_op_zip ~sepfmt op arg (rl1,l2 as zip) =
     or empty lists is returned if the variable should be unset or has an unknown
     previous value. *)
 let reverse_env_update ~sepfmt var op arg cur_value =
-  let arg = transform_format ~sepfmt arg in
-  if String.equal arg  "" && op <> Eq then None else
+  let (arg, _, _) = transform_format ~sepfmt var arg in
```

Comment:
to make the code clearer, I feel like having a record instead of a triple sum type would make things clearer in the future and potentially better at avoiding any future bugs